src/main/scala/com/rawlabs/das/server/cache/iterator/QualSelectivityAnalyzer.scala

```diff
@@ -172,12 +171,52 @@ object QualSelectivityAnalyzer {
   private def valueAsBigDecimal(v: Value): Option[BigDecimal] = {
     if (v == null) return None
     v.getValueCase match {
+      case Value.ValueCase.BYTE    => Some(BigDecimal(v.getByte.getV))
+      case Value.ValueCase.SHORT   => Some(BigDecimal(v.getShort.getV))
       case Value.ValueCase.INT     => Some(BigDecimal(v.getInt.getV))
       case Value.ValueCase.LONG    => Some(BigDecimal(v.getLong.getV))
       case Value.ValueCase.FLOAT   => Some(BigDecimal.decimal(v.getFloat.getV.toDouble))
       case Value.ValueCase.DOUBLE  => Some(BigDecimal.decimal(v.getDouble.getV))
       case Value.ValueCase.DECIMAL => Some(BigDecimal(v.getDecimal.getV))
-      // You could handle BYTE, SHORT, etc. similarly
+      case Value.ValueCase.DATE    =>
+        // Convert date -> days since epoch => BigDecimal
+        val d = v.getDate
+        try {
+          val localDate = LocalDate.of(d.getYear, d.getMonth, d.getDay)
+          // toEpochDay => number of days since 1970-01-01
+          val days = localDate.toEpochDay
+          Some(BigDecimal(days))
+        } catch {
+          case _: Throwable => None
```

Comment:
Throwable is the wrong choice here (and below), as it is way too broad and captures interruption exceptions, out of memory, and other things hat have nothing to do with date conversions. At the very least use NonFatal.

Reply:
Catching DateTimeException now. If ever we'd get wrong numbers from our `Value`, `LocalDateTime` and company would throw.